### PR TITLE
GPIB-343

### PR DESCRIPTION
**Pull request by xiangzhe0812**
Allow Grouped Addresses  on user profile setting.
Allow user put existing address into groupd address.
Allow user put new address into groupd address.

### CODE REVIEW

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM

File: src/components/addresses/AddressForm.js [hunk 1]
```diff
@@ -49,7 +50,7 @@ const AddressForm = ({
       onSubmit={onSubmit}
       enableReinitialize
     >
-      {({ isSubmitting, errors }) => (
+      {({ isSubmitting, errors, setFieldValue }) => (
```
Comment (ahinchliff) [inline src/components/addresses/AddressForm.js]: is `setFieldValue` being used?

File: src/components/addresses/AddressGroupModal.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    history.push(base);
+  };
+
+  const onSubmit = async (values, formActions, modalActions) => {
+    formActions.setSubmitting(true);
+    try {
+      await gpib.secure.put(`/address/group/${id}`, values);
+      await mutate(`/user/${user.id}/address`);
+      modalActions.onDismiss();
+      formActions.setSubmitting(false);
```
Comment (ahinchliff) [inline src/components/addresses/AddressGroupModal.js]: move `formActions.setSubmitting(false)` out of the try catch to reduce duplication

**Review by rellfy — APPROVED** [review]

File: src/pages/AddressesPage.js [hunk 3]
```diff
@@ -8,24 +8,31 @@ import Loader from "components/Loader";
...
+  const [groupAddress, setGroupAddress] = useState([]);
+  const { data: settings, error: fetchSettingsError } = useSWR(
+    `/settings/${user.id}`
+  );
   useEffect(() => {
-    if (!addresses) return;
-    const hasAddress = addresses.find((a) => a.id === selected);
-    if (!hasAddress) setSelected(null);
-  }, [addresses, selected, setSelected]);
+    const groupAddress = addresses?.filter((i) => i.groupID);
```
Comment (ahinchliff) [inline src/pages/AddressesPage.js]: you dont need to do this logic in useEffect. It can be done on each render right?

**Review by ahinchliff — CHANGES_REQUESTED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Pull request merged**